tests/unit/amo/components/TestFeaturedAddons.js

```diff
@@ -6,59 +6,284 @@ import {
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[addonType],
+      },
+      loading: false,
+      results: null,
+    });
+
+    dispatch.reset();
+
+    root.setProps({
+      loading: true,
+      results: null,
+    });
+
+    sinon.assert.notCalled(dispatch);
+  });
+
+  it('does not fetch featured add-ons when results are loaded', () => {
+    const addonType = ADDON_TYPE_EXTENSION;
+    const dispatch = sinon.stub();
+
+    store.dispatch(_loadFeatured());
+    // Get the results props from the state mapper.
+    const mappedProps = mapStateToProps(store.getState());
+
+    const root = render({
+      dispatch,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[addonType],
+      },
+      ...mappedProps,
+      hasSearchParams: true,
+    });
+
+    dispatch.reset();
+
+    root.setProps({
+      // Update an unrelated parameter.
+      hasSearchParams: false,
+    });
+
+    sinon.assert.notCalled(dispatch);
+  });
+
+  it('does not fetch featured add-ons when results are loaded + empty', () => {
+    const addonType = ADDON_TYPE_EXTENSION;
+    const dispatch = sinon.stub();
+
+    store.dispatch(_loadFeatured({ resultMap: {} }));
+    // This will make props.results === []
+    const mappedProps = mapStateToProps(store.getState());
+
+    render({
+      dispatch,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[addonType],
+      },
+      ...mappedProps,
+      hasSearchParams: true,
+    });
+
+    sinon.assert.notCalled(dispatch);
+  });
+
+  it('renders a header without a configured addonType', () => {
+    const root = render({
+      addonType: null,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[ADDON_TYPE_EXTENSION],
+      },
+    });
+
+    expect(root.find('.FeaturedAddons-header'))
+      .toIncludeText('More Featured Extensions');
+  });
+
+  it('renders a header when fetching extensions', () => {
+    store.dispatch(_getFeatured({ addonType: ADDON_TYPE_EXTENSION }));
     const root = render(mapStateToProps(store.getState()));
 
-    expect(root.childAt(0)).toIncludeText('More Featured Extensions');
+    expect(root.find('.FeaturedAddons-header'))
+      .toIncludeText('More Featured Extensions');
   });
 
-  it('renders a FeaturedAddons page with themes HTML', () => {
-    store.dispatch(
-      featuredActions.getFeatured({ addonType: ADDON_TYPE_THEME }));
+  it('renders a header when fetching themes', () => {
+    store.dispatch(_getFeatured({ addonType: ADDON_TYPE_THEME }));
     const root = render(mapStateToProps(store.getState()));
 
-    expect(root.childAt(0)).toIncludeText('More Featured Themes');
+    expect(root.find('.FeaturedAddons-header'))
+      .toIncludeText('More Featured Themes');
+  });
+
+  it('renders a header for a configured addonType', () => {
+    const root = render({
+      addonType: ADDON_TYPE_THEME,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[ADDON_TYPE_EXTENSION],
```

Comment:
This is fine but we do have a helper (`visibleAddonType`) that is probably worth using in case this ever becomes more complex than a mapping.